### DISCUSSION TITLE
1.2.5

### DISCUSSION THREAD

**Discussion by andrey-khropov** [catboost/catboost#2641] [Announcements]
## New features
* \[Python-package\]: Support custom eval metrics on GPU. #1792. Thanks to @pnsemyon.

## Bugfixes
* \[Python-package\]: Check eval_period parameter validity for staged prediction. #2593
* \[Python-package\]: Fix _CustomLoggersStack.pop logic. #2620
* \[R-package\]: Fix Caret object: Inconsistent grid creation with documentation. #2609
* \[JVM applier\]: Fix issues with exposing undesired symbols in JNI shared libraries (including allocators) on macOS. #2606
* Fix training with embedding features on GPU. #2249, #2308, #2591
* Fix training with text features on GPU
* Use correct sample count in MultiRMSE on multiple GPUs. #2557
* Fix sign of 2nd order derivative in Huber loss
* Enable gradient walker for non-additive metrics
* Fixes for Cox objective: buffer overflow in derivatives calculation, derivatives summation, metric calculation, disable ordered boosting
* Fix text features data serialization in the model files

<hr /><em>This discussion was created from the release <a href='https://github.com/catboost/catboost/releases/tag/v1.2.5'>1.2.5</a>.</em>

**Comment by ryanshrott** [comment]
This release seems to have broke text features on GPU for me.
